hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -182,6 +183,13 @@ public static HoodieWriteConfig createHoodieConfig(String schemaStr, String base
       builder = builder.withSchema(schemaStr);
     }
 
+    boolean isSimpleKeyGen = (parameters.getOrDefault(DataSourceWriteOptions.KEYGENERATOR_CLASS_OPT_KEY().key(), DataSourceWriteOptions.DEFAULT_KEYGENERATOR_CLASS_OPT_VAL()))
+        .equals(SimpleKeyGenerator.class.getName());
+    if (isSimpleKeyGen) {
```

Comment:
I chose to serialize the props only when key gen is of type simple. Once we add a validation rule that meta client can be enabled only w/ simple key gen, we can serialize these props only when meta fields are disabled. don't really need to check for simple key gen.